src/ball.js

```diff
@@ -2,59 +2,92 @@ import constrainAngle from './helpers/constrainAngle.js';
 
 class Ball {
   constructor(x, y, radius) {
+    // For scoring
+    this.ballIsHit = false;
+    this.hitCount = 0;
+
     this.radius = radius;
 
     // For detecting if edge of ball off screen in X dir
     this.minX = this.radius;
     this.maxX = width - this.radius;
 
+    // Physics (position & velocity)
+    this.pos = createVector(x, y);
+    this.vel = createVector();
+
     // Constrain angle to always be hitting the ball up
     // ie angleHit = constrain(angleHit, this.minAngle, this.maxAngle)
     this.minAngle = -PI * 0.65;
     this.maxAngle = -PI * 0.35;
 
-    this.pos = createVector(x, y);
-    this.vel = createVector();
-    this.acc = createVector();
-
+    // Magic number constant land:
     this.restitution = 0.8;
-
-    this.gravity = createVector(0, 0.3);
+    this.gravity = createVector(0, 0.3); // Applied as a velocity
     this.speedLimit = 17;
 
-    // For applying user input force to ball
-    this.hitVelocity = createVector();
-    // Magic number, can be adjusted at will until it feels right
+    // How hard does the user hit the ball?
     // No f=ma stuff happening, just applied as a veloctiy (in pixels per frame)
     this.hitMagnitude = this.gravity.y * 100;
+    // For applying hitMagnitude to ball
+    this.hitVelocity = createVector();
+  }
 
-    this.ballIsHit = false;
-    this.hitCount = 0;
+  checkHit(x, y) {
+    // If hit is too high on ball, it will be ignored
+    const tooHigh = this.pos.y - y > this.radius * 0.8;
+
+    const d = dist(this.pos.x, this.pos.y, x, y);
+    const ballIsHit = d < this.radius && !tooHigh;
+
+    // d is being returned for a 'too close to center' check
+    return [ballIsHit, d];
   }
 
-  clickEvent(clickX, clickY) {
-    const d = dist(this.pos.x, this.pos.y, clickX, clickY);
-    this.ballIsHit = d < this.radius;
+  hitAngle(x, y, d) {
+    let angle;
 
-    // If hit is too high on ball, ignore
-    const tooHigh = this.pos.y - clickY > this.radius * 0.8;
+    // How much to deviate from straight up when hit too close to center
+    const dAngle = 0.1;
+    const straightUp = -PI / 2;
 
-    if (this.ballIsHit && !tooHigh) {
+    // If hit too close to center: then direct mostly straight up
+    if (d < this.radius / 3) {
+      angle = random(straightUp - dAngle, straightUp + dAngle);
```

Comment:
Re-did the random angle. before this was `random(-1.6, -1.4)`.  Straight up (in the inverted y graphics world) is at `-PI / 2` (aka about `-1.57`); so the old angle bounds were biased towards the right.